### DISCUSSION TITLE
How to improve the accuracy for large index sizes?

### DISCUSSION THREAD

**Discussion by nithin-nk** [milvus-io/milvus#28010] [Q&A and General discussion]
Hi,
We are using the Milvus index with a large index size (More than 350K documents). We are not getting good accuracy in the results. We were getting very good accuracy with an index size of 50K. How to improve the accuracy for large index sizes?

**Comment by yhmo** [comment]
The recall rate of search mainly depends on index type and index/search parameters. So, what kind of index and parameters are you using.

**Comment by nithin-nk** [comment]
We are using the index type IP and L2. For search parameters we are using the following json
search_params = {
            "IVF_FLAT": {"metric_type": "IP", "params": {"nprobe": 10}},
            "IVF_SQ8": {"metric_type": "IP", "params": {"nprobe": 10}},
            "IVF_PQ": {"metric_type": "IP", "params": {"nprobe": 10}},
            "HNSW": {"metric_type": "IP", "params": {"ef": 10}},
            "RHNSW_FLAT": {"metric_type": "IP", "params": {"ef": 10}},
            "RHNSW_SQ": {"metric_type": "IP", "params": {"ef": 10}},
            "RHNSW_PQ": {"metric_type": "IP", "params": {"ef": 10}},
            "IVF_HNSW": {"metric_type": "IP", "params": {"nprobe": 10, "ef": 10}},
            "ANNOY": {"metric_type": "IP", "params": {"search_k": 10}},
            "AUTOINDEX": {"metric_type": "IP", "params": {}},
        }

**Reply by yhmo** [reply]
IP and L2 is "metric type", not index type.
All the index parameters are listed here: https://milvus.io/docs/index.md

Recall rate is affected by index type and parameters, for example,
- If you created an index IVF_FLAT with parameter 'nlist = 1024', and searched with 'nprobe=10'. The recall rate might be worse than 'nlist=1024' and 'nprobe=200'.
- The recall rate of IVF_PQ/IVF_SQ8 is usually worse than HNSW/IVF_FLAT.

By the way, RHNSW_FLAT/RHNSW_SQ/RHNSW_PQ/ANNOY are no longer supported in latest milvus versions. No need to test them.
